### ISSUE TITLE
assertions in pyunit test cases don't show tracebacks

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#11134]
|[<img alt="radix's avatar" src="https://avatars.githubusercontent.com/u/227068?s=50" width="50" height="50">](https://github.com/radix)| @radix reported|
|-|-|
|Trac ID|trac#1610|
|Type|defect|
|Created|2006-03-29 18:36:53Z|

weird. I imagine it has something to do with the code that strips cruft from tracebacks. Here's an example:

```
import unittest

class FooTest(unittest.TestCase):
    def test_foo(self):
        self.assertEquals(False, True)
```

And the running:

```
radix@Atomsk ~% trial test_crap.py       
Running 1 tests.
test_crap
  FooTest
    test_foo ...                                                         [FAIL]

===============================================================================
[FAIL]: test_crap.FooTest.test_foo

Failure: False != True: exceptions.AssertionError
-------------------------------------------------------------------------------
Ran 1 tests in 0.468s

FAILED (failures=1)
```

whereas if you replace "import unittest" with "from twisted.trial import unittest":

```
radix@Atomsk ~% trial test_crap.py
Running 1 tests.
test_crap
  FooTest
    test_foo ...                                                         [FAIL]

===============================================================================
[FAIL]: test_crap.FooTest.test_foo

  File "/Users/radix/test_crap.py", line 5, in test_foo
    self.assertEquals(False, True)
twisted.trial.unittest.FailTest: False != True
-------------------------------------------------------------------------------
Ran 1 tests in 0.277s

FAILED (failures=1)
```

<details><summary>Searchable metadata</summary>

```
trac-id__1610 1610
type__defect defect
reporter__radix radix
priority__low low
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__trial trial
keywords__ 
time__1143657413000000 1143657413000000
changetime__1149403039000000 1149403039000000
version__ 
owner__ 
cc__spiv
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="radix's avatar" src="https://avatars.githubusercontent.com/u/227068?s=50" width="50" height="50">](https://github.com/radix)|@radix commented|
|-|-|

It's _trimFrame.

**Comment by twisted-trac** [comment]
|[<img alt="radix's avatar" src="https://avatars.githubusercontent.com/u/227068?s=50" width="50" height="50">](https://github.com/radix)|@radix set status to `closed`|
|-|-|

This was fixed in r16988 along with [#7110](https://github.com/twisted/twisted/issues/7110). Thanks jml.

**Comment by twisted-trac** [comment]
|[<img alt="radix's avatar" src="https://avatars.githubusercontent.com/u/227068?s=50" width="50" height="50">](https://github.com/radix)|@radix set status to `reopened`|
|-|-|

never mind, that was reverted.

**Comment by twisted-trac** [comment]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)|@jml set status to `closed`|
|-|-|

Fixed in r16992, see [#7110](https://github.com/twisted/twisted/issues/7110)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun set status to `reopened`|
|-|-|

Re-opened because of revert

**Comment by twisted-trac** [comment]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)|@jml set status to `closed`|
|-|-|

(In [17013]) This time for sure -- Intelligently trim frames

 * Fixes [#7110](https://github.com/twisted/twisted/issues/7110), [#11134](https://github.com/twisted/twisted/issues/11134)
 * Author jml, warner
 * Reviewers exarkun, jml

Check the contents of the frames before we trim them. 
Provide focused unit tests for this.
Include the "Traceback (...)" line in the traceback

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun set status to `reopened`|
|-|-|

Reopened, see comments on [#7110](https://github.com/twisted/twisted/issues/7110)

**Comment by twisted-trac** [comment]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)|@jml set status to `closed`|
|-|-|

(In [17054]) Trim tracebacks intelligently in Trial's output

 * Fixes [#7110](https://github.com/twisted/twisted/issues/7110), [#11134](https://github.com/twisted/twisted/issues/11134)
 * Author: jml, warner
 * Reviewer: jml, spiv

Apathy overwhelms me like a

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
